src/lsp/MessageHandlers.re

```diff
@@ -333,78 +333,92 @@ let handlers: list((string, (state, Json.t) => result((state, Json.t), string)))
       let%opt startPos = PartialParser.positionToOffset(text, start);
       let%opt_wrap endPos = PartialParser.positionToOffset(text, end_);
 
-      let substring = String.sub(text, startPos, endPos - startPos);
-
-      open Utils;
-      let trailingNewlines = substring |> countTrailing('\n');
-      let (leadingNewlines, charsToFirstLines) = {
-        let splitted = substring |> split_on_char('\n');
-        let l = List.length(splitted);
-        let rec loop = (i, leadingLines, skipChars) => {
-          let line = List.nth(splitted, i);
-          switch (line |> String.trim |> String.length) {
-          | 0 => loop(i + 1, leadingLines + 1, skipChars + (line |> String.length))
-          | _ => (leadingLines, skipChars + 1)
+      /* 
+        This is to prevent formatOnPaste 
+        See https://github.com/jaredly/reason-language-server/issues/51 
+        for more information
+      */
+      if (fst(start) == fst(end_) && text.[endPos] == '\n') {
```

Comment:
hmmm did you check that this works? I would think you'd want `text.[endPos] != '\n'`

Reply:
I believe this should be `text.[endPos] == '\n'`. Or just skip this condition all together. 
Reasons:

- You can copy/paste multiple lines
- For a single line, it stills add a new semicolon after pasted content if it's `!= '\n'` 


I mean this is a really annoying "feature" of VSCode. It should give us a head up about formatOnPaste or formatOnSave

Reply:
Here's my desired outcome:

- when I select a whole line and say "format selection", it should refmt it
- when someone pasts in a subsection of a line, it should not be refmt'd
- anything multiline is fair game for refmt'ing. If you have "format on paste" turned on, then that's what you get.

Reply:
Alright. I think I misunderstood the desired behaviour and totally forgot about rangeFormating  the a single line. Fixed it